src/debugger/ios/deviceRunner.ts

```diff
@@ -149,7 +149,8 @@ export class DeviceRunner {
         this.cleanup();
 
         return this.mountDeveloperImage().then(function(): Q.Promise<void> {
-            const {spawnedProcess} = new Node.ChildProcess().spawnWithExitHandler("idevicedebugserverproxy", [proxyPort.toString()]);
+            const spawnInfo = { command: "idevicedebugserverproxy", args: [proxyPort.toString()], waitForExit: true};
+            const {spawnedProcess} = new Node.ChildProcess().spawn(spawnInfo);
             this.nativeDebuggerProxyInstance = spawnedProcess;
             const deferred = Q.defer<ChildProcess>();

```

Comment:
I just saw that this deferred is not actually used, looks like a bug. Apart from that, I think that this is manually implementing the `waitForExit: false` case, so perhaps it should just be returning the outcome promise?